src/components/CodeEditor.services.ts

```diff
@@ -0,0 +1,44 @@
+import { print, parse } from 'graphql';
+
+import { copyToClipBoard, formatQuery, formatJson } from './CodeEditor.utils';
+import { EditorContext, CodeEditorContext } from './CodeEditor.types';
+
+export const copyContext = (context: CodeEditorContext) => {
+  return new Promise((resolve, reject) => {
+    const {
+      external: { selectedRequest, requestMetaDataById },
+      activeContext,
+    } = context;
+
+    try {
+      if (activeContext !== EditorContext.Headers) {
+        copyToClipBoard(
+          activeContext === EditorContext.query
+            ? formatQuery(selectedRequest!.query)
```

Comment:
Is this `!` a TS feature? I tried to find about it, but couldn't find good results without a name or keyword to search for. :joy:

Reply:
would love to know a keyword to search about that too, dunno where I've learned it. but yes, it says to TS compiler to stop complaining about a field that could be undefined if you are sure it would never arrive undefined in that time in the code.

Reply:
Oh, cool! I've been using `as unknown as Something` to accomplish the same thing :joy: